### DISCUSSION TITLE
From XF to Avalonia questions

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [AvaloniaUI/Avalonia#14618] [Q&A] [answered]
Hi,
as XF is about to retire we must decide whats happening with our XF iOS and Android App. We tried to migrate to MAUI, but suspended the migration process. Now we consider alternatives and prefer the C#/.Net way. Our app uses some features and we don't know if Avalonia supports this features. Perhaps somebody can answer our questions:
- Are Firebase Push Notifications (FCM) working?
- It's possible to build iOS Notification Service Extensions?
- Can we listen to native events like OnConfigurationChanged or OnRestoreInstanceState on android?
Thanks

**Comment by jp2masa** [comment, marked as answer]
Avalonia mobile support is currently built on top of Xamarin bindings for Android and iOS APIs. The features you describe are not really UI related. If Xamarin supports them, you can use them with Avalonia as well. From a quick search, they seem to be supported:

- https://learn.microsoft.com/en-us/xamarin/android/data-cloud/google-messaging/firebase-cloud-messaging
- https://learn.microsoft.com/en-us/xamarin/ios/platform/user-notifications/enhanced-user-notifications
- https://learn.microsoft.com/en-us/dotnet/api/android.app.activity.onconfigurationchanged?view=net-android-34.0
- https://learn.microsoft.com/en-us/dotnet/api/android.app.activity.onrestoreinstancestate?view=net-android-34.0

**Reply by maxkatz6** [reply]
Yes, pretty much none of these questions are related to XF specifically, and more about Xamarin or underlying platform, which is the same here.

**Comment by AlleSchonWeg** [comment]
Thank you for the quick response.
I have an additional question:
Has Avalonia the same controls as MAUI/Xamarin? I talk about Switch, Button, NavigationView, TabbedPage  or ListView.

Thanks again.

**Reply by timunie** [reply]
NavigationView --> not build in, but via Awsome-Avalonia there are some 3rd party controls
TabbedPage --> Check Avalonia.Labs 

In general it is easier to add own controls or restyle existing ones in Avalonia imo, as you have the same look and feel on all platforms.
